pkg/didcomm/dispatcher/api.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package dispatcher
+
+import "github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+
+// Service protocol service
+type Service interface {
+	Handle(msg DIDCommMsg) error
+	Accept(msgType string) bool
+	Name() string
+}
+
+// DIDCommMsg did comm msg
+type DIDCommMsg struct {
+	// Outbound indicates the direction of this DIDComm message:
+	//   - outgoing (to another agent)
+	//   - incoming (from another agent)
+	Outbound bool
+	Type     string
+	Payload  []byte
+}
+
+// Destination provides the recipientKeys, routingKeys, and serviceEndpoint populated from Invitation
+type Destination struct {
+	RecipientKeys   []string
+	ServiceEndpoint string
+	RoutingKeys     []string
+}
+
+// Outbound interface
+type Outbound interface {
+	Send([]byte, *Destination) error
```

Comment:
nitpick - I'd appreciate if instead of `[]byte` this was an `interface{}`, thus relieving all callers from having to marshal the messages themselves (there will be several callers that will have to do this)

Reply:
@troyronda @rolsonquadras what do you think?

Reply:
@fqutishat i feel interface{} would be more appropriate.

Reply:
yea i think interface would be more appropriate

Reply:
ok I will change it

Reply:
done